hooks/pre-commit

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+from __future__ import print_function
+from future import standard_library
+standard_library.install_aliases()
+
+import os
+import urllib.request
+import urllib.parse
+import urllib.error
+from distutils.version import LooseVersion
+import re
+import sys
+import subprocess
+
+pattern = re.compile(r'[."]version["\s":=]+(\d+\.\d+.\d+)')
+version_dict = {}
+version_file = urllib.request.urlopen(
+    "https://raw.githubusercontent.com/weecology/retriever/master/version.txt")
+version_file.readline()
+
+
+def to_string(value):
+    if sys.version_info >= (3, 0, 0):
+        return value.decode("UTF-8")
+    else:
+        return value
+
+
+for line in version_file:
+    key, value = to_string(line).split(",")
+    version_dict[key] = value
+
+
+def get_script_version(file_name):
+    file_name = os.path.basename(file_name)
+    return LooseVersion(version_dict[file_name])
+
+
+ # check if upstream is added as remote
+remotes = to_string(subprocess.check_output("git remote -v", shell=True))
+
+if u"upstream\thttps://github.com/weecology/retriever.git (fetch)" not in remotes:
```

Comment:
That knocks a few birds down with one stone! I implemented the changes so that the newer version checks for differences regardless of how their upstream is setup. It seems to be working well.